### PR TITLE
Migrate all Kibana server side APIs to call ES-reporting APIs

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*

*Description of changes:*
#### Major Change
- Add converters between ES(backend) model and UI model
- Refactor every API call related to reports and report definition
  - create/get/delete/update/list...

#### Minor
- Fix in-context menu csv wrong time range parameters
- remove origin from delivery in UI schema, Add to "core_params" in UI schema

#### TODO:
- remove deprecated reports-scheduler related code in kibana-reports plugin
- apply `fromIndex` parameter for `getAll` APIs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/common/index.ts [hunk 1]
```diff
@@ -10,3 +10,14 @@ export const REPORTS_SCHEDULER_API = {
 export const NOTIFICATION_API = {
   SEND: '/_opendistro/_notifications/send',
 };
+
+const BASE_REPORTS_URI = '/_opendistro/_reports';
+
+export const ES_REPORTS_API = {
+  ON_DEMAND_REPORT: `${BASE_REPORTS_URI}/on-demand`,
```
Comment (davidcui-amzn) [inline kibana-reports/common/index.ts:17]: np: change `on-demand` to `on_demand` to maintain consistency

File: kibana-reports/server/backend/opendistro-es-reports-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      },
+    },
+    method: 'GET',
+  });
+
+  esReports.getReports = clientAction({
+    url: {
+      fmt: `${ES_REPORTS_API.LIST_REPORT_INSTANCES}`,
+      //TODO: wrong format error thrown even required = false, need to figure it out the correct setting to make it truly optional
+      // req: {
```
Comment (davidcui-amzn) [inline kibana-reports/server/backend/opendistro-es-reports-plugin.ts:66]: if we have the `TODO` do we also need to still leave in the commented-out code?

File: kibana-reports/server/model/backendModel.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  };
+  format: {
+    duration: string;
+    fileFormat: string;
+    limit?: number;
+    header?: string;
+    footer?: string;
+  };
+  trigger: {
+    triggerType: string; //TODO:
```
Comment (davidcui-amzn) [inline kibana-reports/server/model/backendModel.ts:44]: what's the `TODO` here?

File: kibana-reports/server/model/index.ts [hunk 4]
```diff
@@ -144,13 +152,30 @@ export const channelSchema = schema.object({
   ),
   // TODO: consider add this field next to url-related fields.
   // Need this to build the links in email
-  origin: schema.uri(), //e.g. https://xxxxx.com
+  // origin: schema.uri(), //e.g. https://xxxxx.com
```
Comment (davidcui-amzn) [inline kibana-reports/server/model/index.ts:155]: Can we remove this comment?

File: kibana-reports/server/routes/lib/createReport.ts [hunk 5]
```diff
@@ -54,13 +59,15 @@ export const createReport = async (
   if (savedReportId) {
     reportId = savedReportId;
   } else {
-    const esResp = await saveReport(isScheduledTask, report, esClient);
-    reportId = esResp._id;
+    const esResp = await saveReport(isScheduledTask, report, esReportsClient);
+    // reportId = esResp._id;
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/lib/createReport.ts:63]: Remove commented line

File: kibana-reports/server/routes/lib/createReport.ts [hunk 6]
```diff
@@ -74,6 +81,7 @@ export const createReport = async (
       );
     } else {
       // report source can only be one of [saved search, visualization, dashboard]
+      // Check if security is enabled. TODO: is there a better way to check?
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/lib/createReport.ts:84]: Whats the current check for if security is enabled? the `if` block with the `request.headers.cookie` object?

**Review by davidcui-amzn — APPROVED** [review]
Please address comments before merging

File: kibana-reports/common/index.ts (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/common/index.ts:17]: Agree, but this will require ES-reporting plugin change, could you confirm @akbhatta ?

Comment (akbhatta) [reply kibana-reports/common/index.ts:17]: sure. I will do the change.

File: kibana-reports/server/model/backendModel.ts (hunk 3 above)
Comment (zhongnansu) [reply kibana-reports/server/model/backendModel.ts:44]: removed todo and use BACKEND_TRIGGER_TYPE

File: kibana-reports/server/model/index.ts (hunk 4 above)
Comment (zhongnansu) [reply kibana-reports/server/model/index.ts:155]: will do

File: kibana-reports/server/routes/lib/createReport.ts (hunk 5 above)
Comment (zhongnansu) [reply kibana-reports/server/routes/lib/createReport.ts:63]: sure

File: kibana-reports/server/routes/lib/createReport.ts (hunk 6 above)
Comment (zhongnansu) [reply kibana-reports/server/routes/lib/createReport.ts:84]: From my test, if security is not enabled, `request.header.cookie` will be `undefined`

File: kibana-reports/server/backend/opendistro-es-reports-plugin.ts (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/server/backend/opendistro-es-reports-plugin.ts:66]: This is one of the todos I will address in future PRs

File: kibana-reports/common/index.ts (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/common/index.ts]: Updated reports-scheduler, and addressed comment in the latest commit

**Review by akbhatta — APPROVED** [review]

**Pull request merged**